### PR TITLE
Use 2.8-beta3 toolset compiler

### PR DESCRIPTION
**Pull request by jcouv**
~~Fixes https://github.com/dotnet/roslyn/issues/25816~~

### CODE REVIEW

**Comment by jcouv** [comment]
@VSadov @jaredpar @dotnet/roslyn-infrastructure for review. Thanks

**Comment by jasonmalinowski** [comment]
@jcouv Given we're bumping langver, this would be requiring 15.7 Previews as a base for contribution, correct? If so, this needs updating to the contribution documents in this PR.

**Comment by jasonmalinowski** [comment]
@jcouv, not really. Normally updates like this are most deliberate and are deliberated on carefully. You'll probably want to reach out to the staff mailing list internally because we traditionally have held off doing this until we shipped a more stable preview. If that was a concern, I could see updating the command line compilers but not bumping langver.

File: build/Targets/Packages.props [hunk 1]
```diff
@@ -55,10 +55,10 @@
     <MicrosoftMSXMLVersion>8.0.0.0-alpha</MicrosoftMSXMLVersion>
     <!-- Using a private build of Microsoft.Net.Test.SDK to work around issue https://github.com/Microsoft/vstest/issues/373 -->
     <MicrosoftNETTestSdkVersion>15.6.0-dev</MicrosoftNETTestSdkVersion>
-    <MicrosoftNetCompilersVersion>2.7.0</MicrosoftNetCompilersVersion>
+    <MicrosoftNetCompilersVersion>2.8.0-beta3</MicrosoftNetCompilersVersion>
     <MicrosoftNetRoslynDiagnosticsVersion>2.6.1-beta1-62702-01</MicrosoftNetRoslynDiagnosticsVersion>
     <MicrosoftNetCoreILAsmVersion>2.0.0</MicrosoftNetCoreILAsmVersion>
-    <MicrosoftNETCoreCompilersVersion>2.8.0-beta2-62712-07</MicrosoftNETCoreCompilersVersion>
+    <MicrosoftNETCoreCompilersVersion>2.8.0-beta3</MicrosoftNETCoreCompilersVersion>
```
Comment (jaredpar) [inline build/Targets/Packages.props]: Nice. Finally in sync.

**Comment by jcouv** [comment]
@jasonmalinowski Reverted the LangVersion change.
From the set of features in C# 7.3, we'd probably want ref local re-assignment, but we can wait until 15.7 RTM to do that.

**Comment by jcouv** [comment]
@jasonmalinowski Good to go?

**Pull request merged**